src/main/java/data/TaskManager.java

```diff
@@ -81,25 +80,28 @@ public void deleteTask(LocalDate date, int taskIndex) {
             if (dayTasks.isEmpty()) {
                 tasks.remove(date);
             }
+            System.out.println("Task deleted.");
+        } else {
+            System.out.println("The task you are trying to delete does not exist.");
         }
     }
 
+
     /**
      * Updates a task for a specific date and task index.
      *
-     * @param date The date of the task.
-     * @param taskIndex The index of the task to update.
+     * @param date              The date of the task.
+     * @param taskIndex         The index of the task to update.
      * @param newTaskDescription The updated description of the task.
      * @throws IndexOutOfBoundsException If the task index is out of bounds.
      */
     public static void updateTask(LocalDate date, int taskIndex, String newTaskDescription)
-            throws IndexOutOfBoundsException {
+            throws IndexOutOfBoundsException, TaskManagerException {
         try {
             List<Task> dayTasks = tasks.get(date);
-            boolean dayHasTasks = dayTasks != null;
-            boolean taskIndexExists = taskIndex >= 0 && taskIndex < Objects.requireNonNull(dayTasks).size();
-            assert dayTasks != null;
-            assert taskIndexExists;
+            if (dayTasks == null || taskIndex < 0 || taskIndex >= dayTasks.size()) {
+                throw new TaskManagerException("The task you are trying to update does not exist.");
+            }
```

Comment:
Good job on handling potential exceptions with the existing TaskManagerException class!